### PR TITLE
K8SPG-80 new update logic, code improvements

### PR DESCRIPTION
**Pull request by nonemax**

### CODE REVIEW

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |

commit: https://github.com/percona/percona-postgresql-operator/pull/76/commits/f17e3be47ef925d09d24bbc90c358c20d233e2fb
image: `perconalab/percona-postgresql-operator:PR-76-f17e3be4-pgo-apiserver`

image: `perconalab/percona-postgresql-operator:PR-76-f17e3be4-pgo-event`

image: `perconalab/percona-postgresql-operator:PR-76-f17e3be4-pgo-rmdata`

image: `perconalab/percona-postgresql-operator:PR-76-f17e3be4-pgo-scheduler`

image: `perconalab/percona-postgresql-operator:PR-76-f17e3be4-postgres-operator`

image: `perconalab/percona-postgresql-operator:PR-76-f17e3be4-pgo-deployer`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |

commit: https://github.com/percona/percona-postgresql-operator/pull/76/commits/f17e3be47ef925d09d24bbc90c358c20d233e2fb
image: `perconalab/percona-postgresql-operator:PR-76-f17e3be4-pgo-apiserver`

image: `perconalab/percona-postgresql-operator:PR-76-f17e3be4-pgo-event`

image: `perconalab/percona-postgresql-operator:PR-76-f17e3be4-pgo-rmdata`

image: `perconalab/percona-postgresql-operator:PR-76-f17e3be4-pgo-scheduler`

image: `perconalab/percona-postgresql-operator:PR-76-f17e3be4-postgres-operator`

image: `perconalab/percona-postgresql-operator:PR-76-f17e3be4-pgo-deployer`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |

commit: https://github.com/percona/percona-postgresql-operator/pull/76/commits/f17e3be47ef925d09d24bbc90c358c20d233e2fb
image: `perconalab/percona-postgresql-operator:PR-76-f17e3be4-pgo-apiserver`

image: `perconalab/percona-postgresql-operator:PR-76-f17e3be4-pgo-event`

image: `perconalab/percona-postgresql-operator:PR-76-f17e3be4-pgo-rmdata`

image: `perconalab/percona-postgresql-operator:PR-76-f17e3be4-pgo-scheduler`

image: `perconalab/percona-postgresql-operator:PR-76-f17e3be4-postgres-operator`

image: `perconalab/percona-postgresql-operator:PR-76-f17e3be4-pgo-deployer`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| demand-backup | failed |
| init-deploy | passed |
| data-migration-gcs | passed |
| scaling | passed |
| recreate | passed |
| affinity | passed |
| monitoring | passed |

commit: https://github.com/percona/percona-postgresql-operator/pull/76/commits/4f1cbdb1e92f5b35a1d6376fd3f5ecad15c183bc
image: `perconalab/percona-postgresql-operator:PR-76-4f1cbdb1-pgo-apiserver`

image: `perconalab/percona-postgresql-operator:PR-76-4f1cbdb1-pgo-event`

image: `perconalab/percona-postgresql-operator:PR-76-4f1cbdb1-pgo-rmdata`

image: `perconalab/percona-postgresql-operator:PR-76-4f1cbdb1-pgo-scheduler`

image: `perconalab/percona-postgresql-operator:PR-76-4f1cbdb1-postgres-operator`

image: `perconalab/percona-postgresql-operator:PR-76-4f1cbdb1-pgo-deployer`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| demand-backup | failed |
| init-deploy | passed |
| data-migration-gcs | passed |
| scaling | passed |
| recreate | passed |
| affinity | passed |
| monitoring | passed |

commit: https://github.com/percona/percona-postgresql-operator/pull/76/commits/fd18155da6577dd9be3de3798b78931cd3c2704d
image: `perconalab/percona-postgresql-operator:PR-76-fd18155d-pgo-apiserver`

image: `perconalab/percona-postgresql-operator:PR-76-fd18155d-pgo-event`

image: `perconalab/percona-postgresql-operator:PR-76-fd18155d-pgo-rmdata`

image: `perconalab/percona-postgresql-operator:PR-76-fd18155d-pgo-scheduler`

image: `perconalab/percona-postgresql-operator:PR-76-fd18155d-postgres-operator`

image: `perconalab/percona-postgresql-operator:PR-76-fd18155d-pgo-deployer`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | passed |
| demand-backup | failed |
| data-migration-gcs | passed |
| scaling | passed |
| recreate | passed |
| affinity | passed |
| monitoring | passed |

commit: https://github.com/percona/percona-postgresql-operator/pull/76/commits/eae2b9ad379a27f78f03d932fbc08fa5fcfc4ce6
image: `perconalab/percona-postgresql-operator:PR-76-eae2b9ad-pgo-apiserver`

image: `perconalab/percona-postgresql-operator:PR-76-eae2b9ad-pgo-event`

image: `perconalab/percona-postgresql-operator:PR-76-eae2b9ad-pgo-rmdata`

image: `perconalab/percona-postgresql-operator:PR-76-eae2b9ad-pgo-scheduler`

image: `perconalab/percona-postgresql-operator:PR-76-eae2b9ad-postgres-operator`

image: `perconalab/percona-postgresql-operator:PR-76-eae2b9ad-pgo-deployer`

File: percona/controllers/service/service.go [hunk 1]
```diff
@@ -0,0 +1,230 @@
...
+		if err != nil {
+			return errors.Wrapf(err, "create service %s", service.Name)
+		}
+		return nil
+	}
+	service.Spec.ClusterIP = oldSvc.Spec.ClusterIP
+	if reflect.DeepEqual(service.Spec, oldSvc.Spec) {
+		return nil
+	}
+	service.ResourceVersion = oldSvc.ResourceVersion
```
Comment (qJkee) [inline percona/controllers/service/service.go:62]: resource version is being checked too, please move it before comparing svc specs

**Review by qJkee — APPROVED** [review]

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| demand-backup | failed |
| init-deploy | passed |
| data-migration-gcs | passed |
| scaling | passed |
| recreate | passed |
| affinity | passed |
| monitoring | passed |

commit: https://github.com/percona/percona-postgresql-operator/pull/76/commits/797b6234b5ea7572e909c59f5d8f76c317eef7a0
image: `perconalab/percona-postgresql-operator:PR-76-797b6234-pgo-apiserver`

image: `perconalab/percona-postgresql-operator:PR-76-797b6234-pgo-event`

image: `perconalab/percona-postgresql-operator:PR-76-797b6234-pgo-rmdata`

image: `perconalab/percona-postgresql-operator:PR-76-797b6234-pgo-scheduler`

image: `perconalab/percona-postgresql-operator:PR-76-797b6234-postgres-operator`

image: `perconalab/percona-postgresql-operator:PR-76-797b6234-pgo-deployer`

File: percona/controllers/service/service.go (hunk 1 above)
Comment (nonemax) [reply percona/controllers/service/service.go:62]: `ResourceVersion` is a part of `ObectMeta`, but we comparing `Spec`

**Pull request merged**